src/hooks/useCds/translate.js

```diff
@@ -249,11 +321,52 @@ export function translateResponse(patientData, stridesData) {
   console.log("translate completed.")
 }
 
-function patientDataToHash(patientData) {
-  if (patientData == null || patientData.length === 0) {
-    return {}
+export function translateToggleChange(patientData, toggleStatus) {
+  if (patientData == null || patientData.length == 0) {
+    return;
+  }
+
+  const patient = patientData.find(pd => pd.resourceType == 'Patient');
+  handleToggles(patient, patientData, toggleStatus.isImmunosuppressed, immunosuppressedObservation);
+  handleToggles(patient, patientData, toggleStatus.isPregnant, pregnantObservation);
+  handleToggles(patient, patientData, toggleStatus.isPregnantConcerned, pregnantConcernedObservation);
+  handleToggles(patient, patientData, toggleStatus.isSymptomatic, symptomaticObservation);
+}
+
+function handleToggles(patient, patientData, isChecked, obs) {
+  const newObsId = `new-observation-for-${obs.label.toLowerCase()}`;
+  if (isChecked) {
+    const found = patientData.find(pd => pd.id === newObsId);
+    if (!found) {
+      const newObs = {
+        resourceType: 'Observation',
+        id: newObsId,
+        subject: {
+          reference: `Patient/${patient.id}`
+        },
+        code: obs.code,
+        effectiveDateTime: (new Date()).toISOString()
+      }
+
+      if (obs.valueCodeableConcept) {
+        newObs.valueCodeableConcept = obs.valueCodeableConcept
+      }
+
+      patientData.push(newObs)
```

Comment:
Do we want these new manually added Observations to show up in the Screening and Management History? I think this answer is no, but we should discuss. Right now they show up as Unnamed Observations in the screening and management history:

<img width="583" alt="Manually Added Observations" src="https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/assets/51176978/224b8ab7-abd5-41eb-85c0-055ca43b9926">

Reply:
We discussed over Slack and decided that we do not want to modify the Screening and Management History when a switch is toggled.

Reply:
Addressed in https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/pull/32/commits/c118942345895af43dc0b6e1282ff97665c71d57. However, we still need to figure out how to force the "Pregnant" field to ignore the toggle switch.